### ISSUE TITLE
vim will not correctly processs <F11> or <F12>

### ISSUE THREAD

**Issue by drrob1** [vim/vim#3246]
vim 8.1.204 on win10 amd64.
Start vim, even vim -N -u NONE
Enter insert mode, and hit <F11> or <F12> and strange characters appear on screen

**Comment by k-takata** [comment]
Reproduced on Win32 CUI, not on Win32 GUI and Cygwin.
Typing <kbd>F9</kbd> inputs literal text `<F9>`, but typing <kbd>F11</kbd> inputs 0xCE 0x85.

**Comment by zgpio** [comment]
I also encountered this problem.

**Comment by tonymec** [comment]
* What are `'term'` `'ttybuiltin'` and `<F11>` set to in the problematic console terminal?
```vim
        :verbose set term? ttybuiltin? <F11>?
```
  (with the question marks)
* Does it change anything if you toggle `ttybuiltin` as follows:
```vim
        :set invtbi
```
then retry using the `F11` key?

Best regards,
Tony.

**Comment by zgpio** [comment]
![default](https://user-images.githubusercontent.com/19503791/44185260-d13c3780-a145-11e8-8928-2ff8d39a2624.png)
`:set invtbi` doesn't work.

**Comment by tonymec** [comment]
Well, apparently your terminal sends something else for `<F11>` than what Vim thinks it ought to.

Try adding to your vimrc the following snippet:
```vim
if &term == 'win32'
    set t_F1=<Char-0xCE><Char-0x85>
endif
```
or, if it doesn't work, then replace it (after the place where you set `'encoding'` to UTF-8) by
```vim
if &term == 'win32'
    let &t_F1 = "\u0385"
endif
```
(it is important to use **double** quotes around the \u0385). If it still doesn't work, then we'll have to find out exactly what your terminal sends for `<F11>`. For that, open an empty buffer by `:new`, go to Insert mode, and then hit `Ctrl-Q` followed by `F11` (I'm saying Ctrl-Q rather than Ctrl-V because if you source `mswin.vim` your Ctrl-V will have been remapped to the paste operation while Ctrl-Q, a synonym, will usually have been left untouched). Then, after going back to Normal mode, place the cursor over each of the characters which (hopefully) will have been inserted in the buffer and do `ga` over each of them.

Best regards,
Tony.

**Comment by zgpio** [comment]
Two snippets don't work.
![default](https://user-images.githubusercontent.com/19503791/44195851-dd87bb00-a16c-11e8-8c42-9639327b5e2e.png)
The first char is `<F11>`, second is `<F12>`.

**Comment by chrisbra** [comment]
> if &term == 'win32'
>     set t_F1=`<Char-0xCE><Char-0x85>`
> endif

Does not work.

>if &term == 'win32'
>    let &t_F1 = "\u0385"
>endif

works kind of, but only with the default locale of latin1 (not with utf-8). With Latin 1, pressing F11 results in (starting with vim.exe --clean in a cmd with codepage 850):

![grafik](https://user-images.githubusercontent.com/244927/44195885-aef0ec80-a13a-11e8-8b25-e7d331395d0e.png)

with encoding=utf-8 started with (vim.exe --clean --cmd ":set enc=utf-8"), does not change anything:
![grafik](https://user-images.githubusercontent.com/244927/44195982-f7a8a580-a13a-11e8-8430-38291d025077.png)

I have no clue, what terminal key code the win32 console expects.

**Comment by chrisbra** [comment]
I posted some more information here: https://groups.google.com/d/msg/vim_use/LS3_-hzkQjA/pBKlVa2mBgAJ

**Comment by chrisbra** [comment]
Can you please check, if this is fixed by https://github.com/vim/vim/commit/2bc152ab53c4b01072edf6ec2ff61e504cb03cbe (e.g. patch 8.1.0451)?

**Comment by chrisbra** [comment]
fixed by 2bc152ab53c4b01072edf6ec2ff61e504cb03cbe
